tests/libs/install.test.ts

```
@@ -0,0 +1,1183 @@
+        assertSpyCall(denoRunStub, 0, {
+          args: [{
+            cmd: ["echo", "./data_files/cmdecho/dummy.txt"],
+            stdout: "piped",
+          }],
+        });
+      } finally {
+        kyGetStub.restore();
+        denoRunStub.restore();
+      }
+    });
+
+    it('download the file and verify that the path to the downloaded data, prefixed with "a", is output to standard output and record in dim.json, dim-lock.json', async () => {
+      const denoRunStub = stub(Deno, "run");
+      const kyGetStub = createKyGetStub("dummy");
+      try {
+        createEmptyDimJson();
+        await new InstallAction().execute(
+          { name: "cmdechoa", postProcesses: ["cmd echo a"] },
+          "https://example.com/dummy.txt",
+        );
+        assertEquals(
+          await fileExists(
+            "data_files/cmdechoa/dummy.txt",
+          ),
+          true,
+        );
+        assertSpyCall(denoRunStub, 0, {
+          args: [{
+            cmd: ["echo", "a", "./data_files/cmdechoa/dummy.txt"],
+            stdout: "piped",
+          }],
+        });
+      } finally {
+        kyGetStub.restore();
+        denoRunStub.restore();
+      }
+    });
+
+    it('exit with error when specify "cmd" for "postProcess" and download', async () => {
+      const kyGetStub = createKyGetStub("dummy");
+      try {
+        createEmptyDimJson();
+        await new InstallAction().execute(
+          { name: "cmd", postProcesses: ["cmd"] },
+          "https://example.com/dummy.txt",
+        );
+        assertEquals(
+          await fileExists(
+            "data_files/cmd/dummy.txt",
+          ),
+          true,
+        );
+        assertSpyCall(consoleLogStub, 0, {
+          args: [
+            Colors.red(
+              "No command entered",
+            ),
+          ],
+        });
+      } finally {
+        kyGetStub.restore();
+      }
+    });
+
+    it('exit with error when specify "cmd aaa" for "postProcess" and download', async () => {
```

エラーが発生することを確認していますか？

エラーのテストである場合はエラーメッセージと `exit(1)` の確認、エラーのテストでない場合は `denoRunStub` の確認を行う必要があると思います。

---

このテストケースですが、求めている動作は`exit`ではないですね。

> // Do not `exit` because there are commands that do not correspond when obtained from external dim.json.

一例としてはこんな感じでしょうか。
「ログを出力して、エラーを無視する」が期待する動作なので、その内容で始めています。

`output log and ignore error when specify error command such as "cmd aaa" for "postProcesses"`

---

こちらの指摘事項はOKです。